pkg/controllermanager/apis/config/v1alpha1/defaults_test.go

```diff
@@ -469,6 +469,31 @@ var _ = Describe("Defaults", func() {
 		})
 	})
 
+	Describe("CredentialsBindingControllerConfiguration defaulting", func() {
+		It("should default CredentialsBindingControllerConfiguration correctly", func() {
+			expected := &CredentialsBindingControllerConfiguration{
+				ConcurrentSyncs: ptr.To(DefaultControllerConcurrentSyncs),
+			}
+			SetObjectDefaults_ControllerManagerConfiguration(obj)
+
+			Expect(obj.Controllers.CredentialsBinding).To(Equal(expected))
+		})
+
+		It("should not default fields that are set", func() {
+			obj = &ControllerManagerConfiguration{
+				Controllers: ControllerManagerControllerConfiguration{
+					CredentialsBinding: &CredentialsBindingControllerConfiguration{
+						ConcurrentSyncs: ptr.To(10),
+					},
+				},
+			}
+			expected := obj.Controllers.CredentialsBinding.DeepCopy()
+			SetObjectDefaults_ControllerManagerConfiguration(obj)
+
+			Expect(obj.Controllers.CredentialsBinding).To(Equal(expected))
```

Comment:
Should test the value is not set to the default. Currently, the test will not fail if the default value is changed to 10 and defaulting start overwriting the already set value, with the following change the test should catch this case

```suggestion
						ConcurrentSyncs: ptr.To(DefaultControllerConcurrentSyncs + 1),
					},
				},
			}
			SetObjectDefaults_ControllerManagerConfiguration(obj)

			Expect(obj.Controllers.CredentialsBinding).To(Not(Equal(DefaultControllerConcurrentSyncs)))
```

Reply:
Although I get the point, the defaulting logic can be made to set `DefaultControllerConcurrentSyncs + 1`, i.e. `6` and the described scenario will still not fail. So a random number might a be better option.

Reply:
Or maybe the best thing is to iterate two times with different numbers

Reply:
> Or maybe the best thing is to iterate two times with different numbers

This sound very nice, let's do it.